deepmd/dpmodel/descriptor/dpa4_nn/norm.py

```
@@ -0,0 +1,647 @@
+        self.bias = np.zeros((self.n_focus, self.channels), dtype=prec)
+
+        # === Step 2. Index and Weight Buffers ===
+        self.expand_index = map_degree_idx(self.lmax)
+
+        # Pre-fuse degree balancing and channel averaging into a single weight:
+        #   w_d = 1 / ((2l+1) * (lmax+1) * C)
+        # so that the shared RMS statistic is a single weighted sum without
+        # allocating an intermediate (N, D, F, C) buffer beyond x^2 itself.
+        weights_list = []
+        scale = 1.0 / ((self.lmax + 1) * self.channels)
+        for l in range(self.lmax + 1):
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)